compliance_checker/cf/util.py

```diff
@@ -930,16 +930,30 @@ def get_flag_variables(nc):
     return flag_variables
 
 
+def extract_grid_mapping_names(grid_mapping_string):
+    """
+    Extracts all grid mapping variable names from a grid_mapping string.
+
+    :param str grid_mapping_string: The grid_mapping attribute string
+    :return list[str]: List of grid mapping variable names
+    """
+    return re.findall(r"\b\w+(?=:)|\b\w+(?=\s+\w+:)|\b\w+$", grid_mapping_string)
```

Comment:
Would just `some_grid_mapping:` be accepted under this regex?  Is it a valid `grid_mapping` representation in such a case?

Reply:
this is the documentation text for grid_mapping:
"The attribute takes a string value with two possible formats. In the first format, it is a single word, which names a grid mapping variable. In the second format, it is a blank-separated list of words "<gridMappingVariable>: <coordinatesVariable> [<coordinatesVariable> …​] [<gridMappingVariable>: <coordinatesVariable>…​]" , which identifies one or more grid mapping variables, and with each grid mapping associates one or more coordinatesVariables, i.e. coordinate variables or auxiliary coordinate variables."

the code works for ALL cases: 
- Words followed by : are picked up.
- Words not followed by : but separated by spaces (and at the end) are picked up.
- Single isolated words are picked up.

Reply:
pre-commit.ci autofix